File: programmers/level1/숫자_문자열과_영단어.py
def solution(s):
    answer = ''
    n_list = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    n_dic = dict()
    for i in range(10):
        n_dic[n_list[i]] = str(i)
    word = ''
    for i in range(len(s)):
        if s[i].isdecimal():
            answer += s[i]
        else:
            word += s[i]
            if word in n_dic:
                answer += n_dic[word]
                word = ''
    return int(answer)


def solution2(s):
    n_list = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    answer = s
    for i in range(10):
        answer = answer.replace(n_list[i], str(i))
    return answer


if __name__ == '__main__':
    s = "2three45sixseven"
    print(solution2(s))

File: programmers-cpp/level1/숫자_문자열과_영단어.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(string s) {
    int answer = 0;
    map<string, string> m;
    m["zero"] = "0";
    m["one"] = "1";
    m["two"] = "2";
    m["three"] = "3";
    m["four"] = "4";
    m["five"] = "5";
    m["six"] = "6";
    m["seven"] = "7";
    m["eight"] = "8";
    m["nine"] = "9";
    string curs;
    string result;
    for (int i=0;i<s.length();i++) {
        if(isdigit(s[i]) == 0) {
            curs += s[i];
            if (m[curs] !="") {
                result += m[curs];
                curs = "";
            }
        } else{
            result += s[i];
        }
    }
    return stoi(result);
}